tests/e2e/CMakeLists.txt

```diff
@@ -0,0 +1,225 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.13)
+
+project(az_iot_middleware_freertos_e2e_tests)
+
+if(NOT UNIX)
+  message(FATAL_ERROR "E2E tests must be run on Linux")
+endif()
+
+include(CTest)
+enable_testing()
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -m32")
+
+include_directories(${CMAKE_CURRENT_LIST_DIR}/../config_files)# Include config
+
+add_compile_options(-DprojCOVERAGE_TEST=0 -DMBEDTLS_CONFIG_FILE=\"mbedtls_config.h\")
+
+message("Fetching FreeRTOS - This could take up to a minute or so...")
+include(FetchContent)
+FetchContent_Declare(
```

Comment:
lets keep FreeRTOS dir empty so this fetching is done in one place i.e in root dir cmake

Reply:
Want to make sure we do this pull only at one place